src/main/java/org/folio/rest/impl/InvoiceLineNumberAPI.java

```diff
@@ -0,0 +1,68 @@
+package org.folio.rest.impl;
+
+import java.util.Map;
+
+import javax.ws.rs.core.Response;
+
+import org.folio.rest.RestVerticle;
+import org.folio.rest.annotations.Validate;
+import org.folio.rest.jaxrs.model.InvoiceLineNumber;
+import org.folio.rest.jaxrs.resource.InvoiceStorageInvoiceLineNumber;
+import org.folio.rest.persist.PostgresClient;
+import org.folio.rest.tools.messages.MessageConsts;
+import org.folio.rest.tools.messages.Messages;
+import org.folio.rest.tools.utils.TenantTool;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+
+import static org.folio.rest.persist.HelperUtils.SequenceQuery.GET_IL_NUMBER_FROM_SEQUENCE;
+
+public class InvoiceLineNumberAPI implements InvoiceStorageInvoiceLineNumber {
+
+  private static final Logger log = LoggerFactory.getLogger(InvoiceLineNumberAPI.class);
+  private final Messages messages = Messages.getInstance();
+  
+  @Validate
+	@Override
+	public void getInvoiceStorageInvoiceLineNumber(String invoiceId, String lang, Map<String, String> okapiHeaders,
+	    Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    vertxContext.runOnContext((Void v) -> {
+    	try {
+        String tenantId = TenantTool.calculateTenantId(okapiHeaders.get(RestVerticle.OKAPI_HEADER_TENANT));
+        PostgresClient.getInstance(vertxContext.owner(), tenantId).selectSingle(GET_IL_NUMBER_FROM_SEQUENCE.getQuery(invoiceId),
+        		getILNumberReply -> {
+	        	try {
+	        		if(getILNumberReply.succeeded()) {
+	        			String invoiceLineNumber = getILNumberReply.result().getLong(0).toString();
+	        			log.info("--- InvoiceLineNumberAPI getInvoiceStorageInvoiceLineNumber invoiceLineNumber -- " + invoiceLineNumber);
+	        			asyncResultHandler.handle(Future.succeededFuture(InvoiceStorageInvoiceLineNumber.GetInvoiceStorageInvoiceLineNumberResponse
+	        					.respond200WithApplicationJson(new InvoiceLineNumber().withSequenceNumber(invoiceLineNumber))));
+	        		} else {
+	        			logErrorAndRespond400(asyncResultHandler, getILNumberReply.cause());
+	        		}
+	        	} catch(Exception e) {
```

Comment:
As far as I can see the exception can happen here only if `getILNumberReply.result().getLong(0)` throws exception. Not sure if this might happen...

Reply:
I have updated the code.